src/tests/Interop/COM/Activator/Program.cs

```diff
@@ -32,225 +32,308 @@ static void InvalidInterfaceRequest()
+                Assert.AreEqual(CLASS_E_CLASSNOTAVAILABLE, e.HResult, "Unexpected HRESULT");
+            }
+            else
             {
-                var cxt = new ComActivationContext()
-                {
-                    ClassId = CLSID_NotUsed,
-                    InterfaceId = typeof(IClassFactory).GUID,
-                    AssemblyPath = assemblyBPath,
-                    AssemblyName = "AssemblyB",
-                    TypeName = "ClassFromB"
-                };
-
-                var factory = (IClassFactory)ComActivator.GetClassFactoryForType(cxt);
-
-                IntPtr svrRaw;
-                factory.CreateInstance(null, ref iid, out svrRaw);
-                var svr = (IGetTypeFromC)Marshal.GetObjectForIUnknown(svrRaw);
-                Marshal.Release(svrRaw);
-                typeCFromAssemblyB = (Type)svr.GetTypeFromC();
+                var e = Assert.Throws<NotSupportedException>(
+                    () =>
+                    {
+                        var CLSID_NotRegistered = new Guid("328FF83E-3F6C-4BE9-A742-752562032925"); // Random GUID
+                        var cxt = new ComActivationContext()
+                        {
+                            ClassId = CLSID_NotRegistered,
+                            InterfaceId = typeof(IClassFactory).GUID,
+                            AssemblyPath = @"C:\foo.dll"
+                        };
+                        ComActivator.GetClassFactoryForType(cxt);
+                    },
+                    "Built-in COM has been disabled via a feature switch");
             }
-
-            Assert.AreNotEqual(typeCFromAssemblyA, typeCFromAssemblyB, "Types should be from different AssemblyLoadContexts");
         }
 
-        static void ValidateUserDefinedRegistrationCallbacks()
+        static void ValidateAssemblyIsolation(bool builtInComDisabled=false)
         {
-            Console.WriteLine($"Running {nameof(ValidateUserDefinedRegistrationCallbacks)}...");
-
-            string assemblySubPath = Path.Combine(Environment.CurrentDirectory, "Servers");
-            string assemblyAPath = Path.Combine(assemblySubPath, "AssemblyA.dll");
-            string assemblyBPath = Path.Combine(assemblySubPath, "AssemblyB.dll");
-            string assemblyCPath = Path.Combine(assemblySubPath, "AssemblyC.dll");
-            string assemblyPaths = $"{assemblyAPath}{Path.PathSeparator}{assemblyBPath}{Path.PathSeparator}{assemblyCPath}";
-
-            HostPolicyMock.Initialize(Environment.CurrentDirectory, null);
-
-            var CLSID_NotUsed = Guid.Empty; // During this phase of activation the GUID is not used.
-            Guid iid = typeof(IValidateRegistrationCallbacks).GUID;
-
-            using (HostPolicyMock.Mock_corehost_resolve_component_dependencies(
-                0,
-                assemblyPaths,
-                string.Empty,
-                string.Empty))
+            if(!builtInComDisabled)
```

Comment:
This is a huge amount of test logic duplication that makes understanding what and how something is being tested difficult. We should only be branching on the parts of the test that are actually different.